### DISCUSSION TITLE
Steam Web Session

### DISCUSSION THREAD

**Discussion by quer** [quer/the-steam-awards#40] [Ideas]
Hey, so im work on improving the project, 
where one of the things, can be that if you are already running the bots, on a other project, where you have access to the cookies and SessionID. 
Then you can just be able to use the cookies here. to do the requests, as almost all the modules just use the cookies. 
Note, things like playing games cant be done. as it have to happen to the steam client ( the  socket to steam ) 

So we doent need to start steam, and login, just for getting the cookies, and SessionID.

Is this somefing that YOU can see using. 

One way i can see it being used, is if there is a way to get the cookie and Sessionid from Archi. 
Then if you use that to run all the bot 24/7 or somefing. you can just export the info, into a config file. And just run the module on that. 

PLEASE response oven if you cant see a using for it. i just want to know if it shoud be in my to do list. 

also have a 2 ideer base on this, where it save the last know'en seasionID and cookies. from last run, then just see if thoes still work when running the project next time. if not then just refetch them. read more in comment (https://github.com/quer/the-steam-awards/discussions/40#discussioncomment-1932148)

**Comment by SENPAY98K** [comment]
hmm this would be very helpful but only for those whose running on a vps 24/7.
as you are able to get cookies and SessionID of the instance running on vps which does not change unless steam dead sometime, then keep using it without any limit, as said those using vps/24/7 would benifit most of it.

**Reply by quer** [reply]
you are right, that is why i want to know it any do that, as i do not know if i shoud put my time into adding it.

**Reply by SENPAY98K** [reply]
> you are right, that is why i want to know it any do that, as i do not know if i shoud put my time into adding it.

Yes, also cookies often expire and requesting new() will take extra time, might even log in again
well i have not test it before, so im sure about it..

Also if you are wating for more participant in this thread it will take some time since only people watching/following this project will get email notification

**Reply by quer** [reply]
the way i see it, if it is expired, it will run the normal flow, where it will login to steam, and get the cookies. And then start. Here we will only have used 1 request. before to validate if thay have expired or not. so we only use 1 request more then today. but if it is not expired, we only use 1 request, and doent have to wait for steam to validate login, and wait for steam to send the cookies. Also if you have many account steam will not block the connection, when you login to many accounts. as we now just do request. and only login to steam if wee have the need. 

So i think i most cases it will be faster. 

The plan is just to leave it up here  for some time, and see if more do participant. And then deside later on :)

**Reply by SENPAY98K** [reply]
> the way i see it, if it is expired, it will run the normal flow, where it will login to steam, and get the cookies. And then start. Here we will only have used 1 request. before to validate if thay have expired or not. so we only use 1 request more then today. but if it is not expired, we only use 1 request, and doent have to wait for steam to validate login, and wait for steam to send the cookies. Also if you have many account steam will not block the connection, when you login to many accounts. as we now just do request. and only login to steam if wee have the need.
> 
> So i think i most cases it will be faster.
> 
> The plan is just to leave it up here for some time, and see if more do participant. And then deside later on :)

fair enought :)

**Comment by stickerx** [comment]
A good step and it will be a good project, Just keep in mind the delay of each request and rate limit the time between authorizations Session to avoid problems with steam servers & suspends accounts 

more time more safe!
10-20-30 seconds

**Reply by quer** [reply]
it shoud not change the rate of the requests. Right now, the project login to steam, then get the cookies. and then run the module.
whit this change to will just skip the first part. as we now expect that we have the right Cookies, and just use them.


When thinking about it, maby a other ideer/solution will be to save last know cookies and SessionID. then at the beginning of running a bot, have a request, to see if the cookie and SessionID have expired. And then just there, login to steam to get the cookies agirn.

Iknow if you what somefing like running a game whit `steamUser.gamesPlayed([{ game_id: 730 }]);`. Then just start steam at that point. and keep it online until end, if a other module need somefing from steam. 

That way we only connect to steam when we have to. 
The one downsize i can see, if the cookie expire mid running, then all will fail. 


in the next version i have added a global time between each request. where you can run multi account at once. but it will queue the request up, and run then one at a time. when a min time between each request. 
It can allready be tested now. but i have not added any "how to use" yet.

**Comment by SENPAY98K** [comment]
btw Does making sentry files for accounts help in anyway ? 🤔

**Reply by quer** [reply]
storing the  sentry files, is the same as using the "two_factor_code", so it will not change anythink. 
As you send the sentry files to steam, like the two_factor_code. to login.

from what i know of.

**Reply by SENPAY98K** [reply]
> storing the sentry files, is the same as using the "two_factor_code", so it will not change anythink. As you send the sentry files to steam, like the two_factor_code. to login.
> 
> from what i know of.

Ah i see, i remember that now ahahaha just forget about it

**Comment by HenkerX64** [comment]
Good idea, it would be nice if you can reduce many login/logout requests and use existing session (for example from archi and/or storage).
In most module cases I only work with websession and it can speed up user switching

**Reply by quer** [reply]
Thanks for the feed back. by allready having the session and cookies. it will speed the hole progress up, a lot. as the hole login to "steam" can be skipped. unless you have to play a game or somefing like that :)

**Comment by stickerx** [comment]
@quer 

hi quer is there a module to get this badge 
[[PSA] Steam Replay 2022 - Profile Badge (50 XP)] 

https://store.steampowered.com/replay

**Reply by quer** [reply]
i did not see that it was added to steam. But i will create one. now

**Reply by quer** [reply]
it have now been added, look on the frontpage, what module to use
